openedx/core/djangoapps/user_api/views.py

```diff
@@ -341,6 +342,13 @@ def post(self, request):
         # for TOS, privacy policy, etc.
         if data.get("honor_code") and "terms_of_service" not in data:
             data["terms_of_service"] = data["honor_code"]
+        password = data.get("password")
+        password_copy = data.get("password_copy")
+        if password and password_copy and password != password_copy:
```

Comment:
How come I don't see this check is gated by the setting `REGISTRATION_EXTRA_FIELDS['password_copy']=='required'`? Does this mean, the password confirmation logic always applies in the new registration view?